# Import Libraries 

In [2]:
import pandas as pd
import numpy as np
import scipy.stats as stats

# Creating DataFrame

In [4]:
df = pd.DataFrame({"Transaction_Amount":[100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]})

In [5]:
df

Transaction_Amount
0                  100
1                  150
2                   50
3                  100
4                  130
5                  120
6                  100
7                   85
8                   70
9                  150
10                 150
11                 120
12                  50
13                 100
14                 100
15                 140
16                  90
17                 150
18                  50
19                  90
20                 120
21                 100
22                 110
23                  75
24                  65

# Measures of Central Tendency

In [7]:
# Measures of Central Tendency (Mean, Median, Mode)
mean_val = df['Transaction_Amount'].mean()
median_val = df['Transaction_Amount'].median()
mode_val = df['Transaction_Amount'].mode()

print("Mean:", mean_val)
print("Median:", median_val)
print("Modus:", mode_val.tolist())


Mean: 102.6
Median: 100.0
Modus: [100]


## Analisis
- Sebagian besar transaksi cenderung berada di sekitar nilai 100.
- Mean sedikit lebih besar dari median, kemungkinan distribusi miring ke kanan (right-skewed)
- Terdapat beberapa transaksi dengan jumlah yang lebih tinggi dari mayoritas
- Median dan Modus sama, artinya meskipun ada kenaikan di beberapa transaksi, mayoritas penjualan masih stabil di angka 100.

# Measures of Variability 

In [10]:
# Measures of Variability (range, variance, Standard Deviation, min&max, Quantile (1,2,3))

max_val = df['Transaction_Amount'].max()
min_val = df['Transaction_Amount'].min()
range_val = df['Transaction_Amount'].max() - df['Transaction_Amount'].min()
variance_val = df['Transaction_Amount'].var()
std_dev = df['Transaction_Amount'].std()
Q1 = df['Transaction_Amount'].quantile(0.25)
Q2 = df['Transaction_Amount'].median()
Q3 = df['Transaction_Amount'].quantile(0.75)

print("Max :", max_val)
print("Min :", min_val)
print("Range :", range_val)
print("Variance :", variance_val)
print("Standard Deviations :", std_dev)
print("Quantile 0.25 (Q1):", Q1)
print("Quantile 0.5 (Median):", Q2)
print("Quantile 0.75 (Q3):", Q3)

Max : 150
Min : 50
Range : 100
Variance : 1012.7499999999997
Standard Deviations : 31.82373328193912
Quantile 0.25 (Q1): 85.0
Quantile 0.5 (Median): 100.0
Quantile 0.75 (Q3): 120.0


## Analisa
Sebaran data cukup luas, dengan standar deviasi menunjukkan bahwa nilai transaksi bervariasi sekitar ±31.82 dari rata-rata. IQR (35) yang relatif kecil dibandingkan range menunjukkan bahwa sebagian besar data berada pada rentang 85–120, sementara nilai ekstrem berada di luar rentang tersebut dan hanya memengaruhi sebagian kecil data.

# H0 dan H1
H0 = Tidak ada perubahan rata-rata transaksi μ = 100

H1 = Ada peningkatan rata - rata transaksi μ > 100

# T-Test

In [14]:
mu_0 = 100
alpha = 0.05

t_stat, p_val = stats.ttest_1samp(df['Transaction_Amount'], mu_0)

print("T-statistic:", t_stat)
print("P-value:", p_val)

if p_val < alpha:
    print("Hasil: Tolak H0 — Ada peningkatan penjualan.")
else:
    print("Hasil: Gagal tolak H0 — Tidak ada peningkatan penjualan.")

T-statistic: 0.4085001556802841
P-value: 0.6865284813438117
Hasil: Gagal tolak H0 — Tidak ada peningkatan penjualan.


## Hasil T-Test
- T-statistic: 0.4085
- P-value: 0.686
- Alpha: 0.05 (5%)

Dengan p-value = 0.6865 > 0.05, hasil T-Test menunjukkan bahwa tidak ada perbedaan yang signifikan secara statistik antara rata-rata transaksi setelah training dan target awal sebesar 100.